tests/range/range_api.cpp

```diff
@@ -212,6 +212,14 @@ class test_range {
   }
 };
 
+void test_empty_kernel_1d_range(size_t N, sycl::queue q) {
+  int *a = sycl::malloc_shared<int>(q, 1);
+  a[0] = 0;
+  q.parallel_for(sycl::range<1>(N), [=](auto i) { a[0] = 1; }).wait_and_throw();
+  CHECK_VALUE_SCALAR(log, a[0], 1);
+  sycl::free(a, q);
+}
+
```

Comment:
```suggestion
void test_empty_kernel_1d_range(size_t N, sycl::queue q) {
  q.parallel_for(sycl::range<1>(N), [=](auto i) { }).wait_and_throw();
}
```
The name of the function is misleading. The kernel is not empty. 
What the intent here? I don't think current test gives valuable information.

Reply:
Good catch. Before, It was an empty kernel, and then I think more and found this trick... Will fix the name, sorry!

Reply:
NOTE: "this trick" tells you that at least one work-item was executed. Don't you expect all N to be executed? It's not checked by this kernel.

Reply:
Why do you prefer `q.parallel_for(sycl::range<1>(N), [=](auto i) { a[0] = 1; }).wait_and_throw();` over an empty kernel?

Reply:
I was afraid that a smart compiler could optimize away an empty kernel submission...  Where I have more hope that t no the compiler will be able to optimize away `N-1` work-item submission.